apps/web/src/services/dashboards/shared/dashboard-widget-input-form/dashboard-widget-options-form/DashboardWidgetOptionDropdown.vue

```diff
@@ -2,129 +2,311 @@
 import { computed, reactive, watch } from 'vue';
 
 import {
+    getTextHighlightRegex,
     PFieldGroup, PIconButton, PSelectDropdown, PToggleButton,
 } from '@spaceone/design-system';
 import type {
     AutocompleteHandler,
     SelectDropdownMenuItem,
 } from '@spaceone/design-system/types/inputs/dropdown/select-dropdown/type';
+import { get } from 'lodash';
 
 import { i18n } from '@/translations';
 
+import type { VariableModelConfig } from '@/lib/variable-models';
+import { VariableModel } from '@/lib/variable-models';
+import type { IBaseVariableModel } from '@/lib/variable-models/_base/types';
+import { getVariableModelMenuHandler } from '@/lib/variable-models/variable-model-menu-handler';
+
 import type { DashboardVariablesSchema } from '@/services/dashboards/config';
 import DashboardCostWidgetValueOptionDropdown
     from '@/services/dashboards/shared/dashboard-widget-input-form/dashboard-widget-options-form/DashboardCostWidgetValueOptionDropdown.vue';
+import { useWidgetFormStore } from '@/services/dashboards/shared/dashboard-widget-input-form/widget-form-store';
 import type {
     InheritanceMode,
     WidgetOptionsSchemaProperty,
 } from '@/services/dashboards/widgets/_configs/widget-options-schema';
-import { COST_VALUE_WIDGET_OPTION_CONFIGS } from '@/services/dashboards/widgets/_configs/widget-options-schema';
+import {
+    COST_VALUE_WIDGET_OPTION_CONFIGS,
+} from '@/services/dashboards/widgets/_configs/widget-options-schema';
+import { getWidgetOptionKeyByVariableKey } from '@/services/dashboards/widgets/_helpers/widget-schema-helper';
 
 const props = defineProps<{
-    schemaKey?: string;
-    label: string;
-    selected?: SelectDropdownMenuItem[];
-    selectionType?: WidgetOptionsSchemaProperty['selection_type'];
-    inherit?: boolean;
-    inheritanceMode?: InheritanceMode;
-    deletable?: boolean;
-    menuHandlers?: AutocompleteHandler[];
+    propertyName: string;
     variablesSchema?: DashboardVariablesSchema;
+    isValid?: boolean;
 }>();
-const emit = defineEmits<{(e: 'update:inherit', isInherit: boolean): void;
-    (e: 'update:selected', selected: SelectDropdownMenuItem[]): void;
+const emit = defineEmits<{(e: 'update:is-valid', isValid: boolean): void;
     (e: 'delete'): void;
 }>();
```

Comment:
praise: Oh, you moved option updating logic to here! I think it is far better than the version before (that I wrote 😅).
It's more clear since each concern is separated appropriately...!!! 
- dropdown: updating logics for each option
- form: logics that must be calculated by all options